Source/WTF/wtf/MathExtras.h

```diff
@@ -771,6 +776,73 @@ inline bool isRepresentableAs(double value)
     return isRepresentableAsImpl<ResultType, double, int64_t>(value);
 }
 
+template<typename T>
+ALWAYS_INLINE constexpr T roundUpToMultipleOfImpl(size_t divisor, T x)
+{
+    T remainderMask = static_cast<T>(divisor) - 1;
+    return (x + remainderMask) & ~remainderMask;
+}
+
+// Efficient implementation that takes advantage of powers of two.
+template<typename T>
+inline constexpr T roundUpToMultipleOf(size_t divisor, T x)
+{
+    ASSERT_UNDER_CONSTEXPR_CONTEXT(divisor && isPowerOfTwo(divisor));
+    return roundUpToMultipleOfImpl<T>(divisor, x);
+}
+
+template<size_t divisor> constexpr size_t roundUpToMultipleOf(size_t x)
+{
+    static_assert(divisor && isPowerOfTwo(divisor));
+    return roundUpToMultipleOfImpl(divisor, x);
+}
+
+template<size_t divisor, typename T> inline constexpr T* roundUpToMultipleOf(T* x)
+{
+    static_assert(sizeof(T*) == sizeof(size_t));
+    return reinterpret_cast<T*>(roundUpToMultipleOf<divisor>(reinterpret_cast<size_t>(x)));
+}
+
+template<typename T>
+inline constexpr T roundUpToMultipleOfNonPowerOfTwo(size_t divisor, T x)
```

Comment:
These function names are super confusing! We should probably rename these functions (in a separate PR).
Maybe `roundUpToMultipleOfPowerOfTwo` and `roundUpToMultipleOf`.